presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryProvider.java

```diff
@@ -51,9 +55,22 @@ public boolean isBufferOwner()
         }
     }
 
+    static class DictionaryEntry
+    {
+        private final long[] values;
+        private final int size;
+
+        DictionaryEntry(long[] values, int size)
+        {
+            this.values = requireNonNull(values);
+            this.size = size;
```

Comment:
It could be. I'm slightly concerned about the amount of double checking we are doing. DictionaryBuffer resizing logic is handled inside the class and tested, and the long input stream logic should also be well tested. I think they should have a little bit of branching cost right?

Reply:
Since this is a private class, if we are not checking this is ok.